src/components/Explore/TracesByService/TracesByServiceScene.tsx

```diff
@@ -73,36 +73,42 @@ export class TracesByServiceScene extends SceneObjectBase<TraceSceneState> {
 
     const exploration = getTraceExplorationScene(this);
     const metricVariable = exploration.getMetricVariable();
-    metricVariable.subscribeToState((newState, prevState) => {
-      if (newState.value !== prevState.value) {
-        const selection = getDefaultSelectionForMetric(newState.value as MetricFunction);
-        if (selection) {
-          this.setState({ selection });
+    this._subs.add(
+      metricVariable.subscribeToState((newState, prevState) => {
+        if (newState.value !== prevState.value) {
+          const selection = getDefaultSelectionForMetric(newState.value as MetricFunction);
+          if (selection) {
+            this.setState({ selection });
+          }
+          this.updateQueryRunner(newState.value as MetricFunction);
+          this.updateBody();
         }
-        this.updateQueryRunner(newState.value as MetricFunction);
-        this.updateBody();
-      }
-    });
+      })
+    );
 
-    this.subscribeToState((newState, prevState) => {
-      const timeRange = sceneGraph.getTimeRange(this);
-      const selectionFrom = newState.selection?.timeRange?.from;
-      // clear selection if it's out of time range
-      if (selectionFrom && selectionFrom < timeRange.state.value.from.unix()) {
-        this.setState({ selection: undefined });
-      }
+    this._subs.add(
+      this.subscribeToState((newState, prevState) => {
```

Comment:
There may be an obvious answer here but why do we have to `this.subscribeToState` and add it to `this._subs.add`?